### PR TITLE
feat: support event with callback

### PR DESCRIPTION
**Pull request by qingzhuozhen**
### Summary
add backs on event level

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Java/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?: no

### CODE REVIEW

File: src/main/java/com/amplitude/Amplitude.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import java.util.*;
 import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Future;
```
Comment (yuhao900914) [inline src/main/java/com/amplitude/Amplitude.java:5]: Why do we need to import this?

Comment (qingzhuozhen) [reply src/main/java/com/amplitude/Amplitude.java:5]: Good catch, let me remove it. This was added when I play around the suggestion to return a Future and later we picked another way.

File: .github/workflows/pull-request-test.yml [hunk 2]
```diff
@@ -4,13 +4,14 @@ on: [pull_request]
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    runs-on: macos-latest
```
Comment (justin-fiedler) [inline .github/workflows/pull-request-test.yml:7]: Do we need macos? [Costs 10x](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates) compared to Ubuntu.

File: src/main/java/com/amplitude/HttpTransport.java [hunk 3]
```diff
@@ -311,11 +311,14 @@ protected boolean shouldRetryForStatus(Status status) {
   }
 
   private void triggerEventCallbacks(List<Event> events, int status, String message) {
-    if (callbacks == null || events == null || events.isEmpty()) {
+    if (events == null || events.isEmpty()) {
       return;
     }
     for (Event event : events) {
-      callbacks.onLogEventServerResponse(event, status, message);
+      AmplitudeCallbacks eventCallback = event.callback != null ? event.callback : callbacks;
```
Comment (justin-fiedler) [inline src/main/java/com/amplitude/HttpTransport.java:318]: I think should we still call the client level callbacks here in addition to the per-event callback

**Review by justin-fiedler — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request-test.yml (hunk 2 above)
Comment (qingzhuozhen) [reply .github/workflows/pull-request-test.yml:7]: That is copied from @bohan-amplitude 's change to make the tests more stable.. @bohan-amplitude maybe see if we can make this more stable with ubuntu?

File: src/main/java/com/amplitude/HttpTransport.java (hunk 3 above)
Comment (qingzhuozhen) [reply src/main/java/com/amplitude/HttpTransport.java:318]: We can do that too, I was trying to debate if we need to do replace or in addition. Would in addition sounds more natural for the intention?

File: .github/workflows/pull-request-test.yml (hunk 2 above)
Comment (qingzhuozhen) [reply .github/workflows/pull-request-test.yml:7]: let revert this for now.

File: src/main/java/com/amplitude/HttpTransport.java (hunk 3 above)
Comment (qingzhuozhen) [reply src/main/java/com/amplitude/HttpTransport.java]: Updated to call extra now.

**Review by bohan-amplitude — APPROVED** [review]

**Pull request merged**